core/src/main/scala/org/apache/spark/deploy/JsonProtocol.scala

```diff
@@ -102,8 +102,8 @@ private[deploy] object JsonProtocol {
     ("name" -> obj.desc.name) ~
     ("cores" -> obj.coresGranted) ~
     ("user" -> obj.desc.user) ~
-    ("memoryperslave" -> obj.desc.memoryPerExecutorMB) ~
-    ("resourcesperslave" -> obj.desc.resourceReqsPerExecutor
+    ("memoryperexecutor" -> obj.desc.memoryPerExecutorMB) ~
```

Comment:
This is the only kind of thing where we might run into problems, if external tools are expecting the old key. Can we output both for the foreseeable future? I am not sure whether this JSON is 'externalized' or purely internal.

Reply:
Sure I think outputting both for the next while is ok. I'll add a comment in the Json protocol that says it's deprecated.